src/blueapi/service/main.py

```diff
@@ -304,6 +304,7 @@ def get_tasks(
 @start_as_current_span(TRACER, "task.task_id")
 def set_active_task(
     task: WorkerTask,
+    instrument_session: str,
```

Comment:
Could: There is an ongoing [discussion](https://github.com/DiamondLightSource/dodal/issues/991#issuecomment-2637883125) about whether the session should be a required parameter. I'm still on the side of making it optional, but either way the discussion should be resolved and actioned before this is merged.

Reply:
Does the PathProvider clear its instrument_session after a task has been completed, to allow us to except when there is no instrument_session passed and we expect there to be?

Reply:
I presume blueapi would have to clear it at the beginning of any task, whether or not a session is passed, to make that consistent.